### PR TITLE
feat: formalize PR lifecycle controller FSM

### PR DESCRIPTION
**Pull request by H31MDALLR**
# feat: formalize PR lifecycle controller FSM

## Overview

Extract the PR lifecycle controller status contract into a dedicated FSM
module and wire the controller through that contract for status validation and
transition execution.

## Motivation

The controller already behaved like a state machine, but the transition policy
was embedded in controller code. This slice makes the lifecycle contract
explicit, testable, and reusable through the shared FSM foundation.

## Changes In Detail

- Add `pr_lifecycle.fsm` with:
  - controller status order and transition graph loaded from
    `config/pr-lifecycle/fsm.edn`
  - valid-status predicates and valid-target helpers
  - explicit transition execution with structured failure maps
  - compiled controller machine config backed by `fsm/define-machine`
- Refactor `pr_lifecycle.controller` to route `update-status!` through the FSM
  and preserve the existing exception/reporting contract
- Load controller defaults from
  `config/pr-lifecycle/controller-defaults.edn`
- Localize controller/FSM emitted text through
  `config/pr-lifecycle/messages/en-US.edn`
- Update controller tests to assert rejected invalid transitions, terminal
  behavior, and idempotent same-state updates
- Add focused FSM tests for the happy path, fix loop, invalid inputs, terminal
  states, and valid-target predicates

## Testing Plan

- `clojure -M:test -e '(require ... ) (clojure.test/run-tests ...)'` for the
  `pr-lifecycle.fsm-test` and `pr-lifecycle.controller-test` namespaces
- `clj-kondo --lint` on the touched `pr-lifecycle` source and test files

## Checklist

- [x] PR lifecycle status transitions are FSM-backed
- [x] Invalid transitions are rejected with structured errors
- [x] Tests updated alongside the controller change
- [x] Targeted lint passes
- [x] Targeted namespace tests pass

### CODE REVIEW

File: components/pr-lifecycle/src/ai/miniforge/pr_lifecycle/fsm.clj [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+  [from-status to-status]
+  (cond
+    (not (valid-status? from-status))
+    (transition-failure :invalid-state from-status to-status)
+
+    (not (valid-status? to-status))
+    (transition-failure :invalid-target-status from-status to-status)
+
+    (= from-status to-status)
+    {:success? true
```
Comment (H31MDALLR) [inline components/pr-lifecycle/src/ai/miniforge/pr_lifecycle/fsm.clj]: Use our success and anomaly factory function patterns

File: components/pr-lifecycle/src/ai/miniforge/pr_lifecycle/controller.clj [hunk 2]
```diff
@@ -57,52 +129,48 @@
...
-         :review-monitor nil
-         :fix-iterations 0
-         :ci-retries 0
-         :history []
-         :created-at (java.util.Date.)
-         :updated-at (java.util.Date.)}))
+   & {:as opts}]
+  (let [defaults (merge (controller-config/controller-defaults)
+                        {:merge-policy merge/default-merge-policy}
+                        opts)
```
[the remark this answers is not in the document]
Comment (H31MDALLR) [reply components/pr-lifecycle/src/ai/miniforge/pr_lifecycle/controller.clj]: Fixed in dc2c74d3. `create-controller` now drops nil-valued opts before merging with the EDN defaults, so explicit `nil` no longer wipes out configured poll intervals or booleans. I also added a regression test covering nil option overrides.

File: components/pr-lifecycle/src/ai/miniforge/pr_lifecycle/fsm.clj [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+                                {:resource fsm-config-resource})
+                    {:hint (messages/t :config/classpath-hint)}))))
+
+(def ^:private fsm-config
+  (delay (load-fsm-config)))
+
+(def initial-status
+  "Initial controller status."
+  (get-in @fsm-config [:pr-lifecycle/fsm :initial-status]))
+
```
[the remark this answers is not in the document]
Comment (H31MDALLR) [reply components/pr-lifecycle/src/ai/miniforge/pr_lifecycle/fsm.clj:67]: Fixed in dc2c74d3. I removed the redundant `delay` and made the eager config load explicit, since this namespace already materializes the FSM config at require time for the exported machine constants.
